component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/handler/DPoPAuthenticationHandler.java

```diff
@@ -0,0 +1,343 @@
+
+                    MDC.put(DPoPConstants.SERVICE_PROVIDER, serviceProvider);
+                    // Set OAuth service provider details to be consumed by the provisioning framework.
+                    setProvisioningServiceProviderThreadLocal(clientApplicationDTO.getConsumerKey(),
+                            serviceProviderTenantDomain);
+
+                } else {
+                    String errorMessage = String.format("Error occurred while trying to authenticate." +
+                            "The %s header value is not defined correctly.", DPoPConstants.OAUTH_DPOP_HEADER);
+                    log.error(errorMessage);
+                    throw new AuthenticationFailException(errorMessage);
+                }
+            }
+        }
+        return authenticationResult;
+    }
+
+    @Override
+    public int getPriority(MessageContext messageContext) {
+
+        return getPriority(messageContext, 24);
+    }
+
+    @Override
+    public boolean canHandle(MessageContext messageContext) {
+
+        return AuthConfigurationUtil.isAuthHeaderMatch(messageContext, DPoPConstants.OAUTH_DPOP_HEADER) ||
+                AuthConfigurationUtil.isAuthHeaderMatch(messageContext, DPoPConstants.OAUTH_HEADER);
+    }
+
+    private AuthenticationResult getAuthenticationResult(AuthenticationResult authenticationResult,
+                                                         OAuth2TokenValidationResponseDTO responseDTO,
+                                                         String authorizationHeader,
+                                                         AuthenticationRequest authenticationRequest,
+                                                         MessageContext messageContext, String accessToken,
+                                                         String consumerKey)
+            throws AuthenticationFailException {
+
+        if (!responseDTO.isValid()) {
+            log.debug(responseDTO.getErrorMsg());
+            return authenticationResult;
+        }
+
+        TokenBinding binding = responseDTO.getTokenBinding();
+        if (binding != null && DPoPConstants.OAUTH_DPOP_HEADER.equals(binding.getBindingType())) {
+            if (!authorizationHeader.startsWith(DPoPConstants.OAUTH_DPOP_HEADER)) {
+                log.debug("DPoP prefix is not defined correctly in the Authorization header.");
+                return authenticationResult;
+            }
+            String dpopHeader = authenticationRequest.getHeader(DPoPConstants.OAUTH_DPOP_HEADER);
+
+            if (StringUtils.isBlank(dpopHeader)) {
+                log.debug("DPoP header is empty.");
+                return authenticationResult;
+            }
+            try {
+                String thumbprintOfPublicKey = Utils.getThumbprintOfKeyFromDpopProof(dpopHeader);
+                if (StringUtils.isBlank(thumbprintOfPublicKey)) {
+                    log.debug("Thumbprint value of the public key is empty in the DPoP Proof.");
```

Comment:
For debug logs, check log.isdebugEnabled()

Reply:
Fixed